omnn/math/Valuable.cpp

```diff
@@ -1801,8 +1801,12 @@ bool Valuable::SerializedStrEqual(const std::string_view& s) const {
     {
         if (exp)
             return exp->operator<(v);
+        // Handle NaN cases first (IEEE 754-2019 section 6.2.1)
+        // "When either operand of a relational operation is NaN, the result is false"
+        if (IsNaN() || v.IsNaN())
+            return false;
```

Comment:
It doesn't look enough information in the context to state that NaN is not less then v nor otherwise. Don't code something nobody can be sure about. This project is about total consolidability of multilayered expressions to build reliable logic upon that. Consider updating documentation to outline the guideline.